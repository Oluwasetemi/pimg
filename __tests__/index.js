const React = require('react')
const { configure, mount } = require('enzyme')
const Adapter = require('enzyme-adapter-react-16')
import EnzymeToJSON from 'enzyme-to-json'
import { setTimeout } from 'timers'

// Setup Enzyme Adapter
configure({ adapter: new Adapter() })

const Image = require('../dist/pimg')

describe('pimg', () => {
	test('basic render', () => {
		const image =
			'https://res.cloudinary.com/stackpie/image/upload/v1513979515/-895520106_m1whb3.jpg'

		const thumb = image.replace('/upload/', '/upload/t_media_lib_thumb/')

		const wrapper = mount(<Image src={image} />)

		expect(wrapper.props().src).toBe(image)

		if (wrapper.state().loading) {
			expect(wrapper.state().thumb).toBe(thumb)
		}

		expect(EnzymeToJSON(wrapper)).toMatchSnapshot()
	})
})
