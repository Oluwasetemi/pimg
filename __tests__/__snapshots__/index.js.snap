// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image can take a placeholder prop 1`] = `
<Image
  dataSaver={false}
  placeholder="http://myPlaceholder"
  src="https://res.cloudinary.com/stackpie/image/upload/v1513979515/-895520106_m1whb3.jpg"
>
  <img
    className="pimg pimg__loading"
    src="http://myPlaceholder"
  />
</Image>
`;

exports[`pimg renders correctly 1`] = `
<Image
  dataSaver={false}
  src="https://res.cloudinary.com/stackpie/image/upload/v1513979515/-895520106_m1whb3.jpg"
>
  <img
    className="pimg pimg__loading"
    src="https://res.cloudinary.com/stackpie/image/upload/c_thumb,w_30/v1513979515/-895520106_m1whb3.jpg"
  />
</Image>
`;
